packages/@n8n/di/src/di.ts

```diff
@@ -11,13 +11,16 @@ type AbstractConstructable<T = unknown> = abstract new (...args: unknown[]) => T
 
 type ServiceIdentifier<T = unknown> = Constructable<T> | AbstractConstructable<T>;
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type Factory<T = unknown> = (...args: any[]) => T;
```

Comment:
Is there a reason we need `any` here?
```suggestion
type Factory<T = unknown> = (...args: unknown[]) => T;
```